crystal_diffusion/analysis/dataset_analysis.py

```diff
@@ -0,0 +1,111 @@
+"""Diffusion Dataset analysis.
+
+This script computes and plots different features of a dataset used to train a diffusion model.
+"""
+import os
+from typing import Dict
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+from crystal_diffusion import ANALYSIS_RESULTS_DIR, DATA_DIR
+from crystal_diffusion.analysis import PLEASANT_FIG_SIZE, PLOT_STYLE_PATH
+from crystal_diffusion.data.parse_lammps_outputs import parse_lammps_output
+
+DATASET_NAME = 'si_diffusion_v1'
+
+
+def compute_metrics_for_a_run(run_path: str) -> Dict[str, pd.Series]:
+    """Get the energy, forces average, RMS displacement and std dev for a single MD run.
+
+    Args:
+        run_path: path to LAMMPS output directory. Should contain a dump file and a thermo log file.
+
+    Returns:
+        metrics evaluated at each MD step organized in a dict
+    """
+    dump_file = [d for d in os.listdir(run_path) if 'dump' in d]
+    thermo_file = [d for d in os.listdir(run_path) if 'thermo' in d]
+
+    df = parse_lammps_output(os.path.join(run_path, dump_file[0]), os.path.join(run_path, thermo_file[0]), None)
+    metrics = {}
+    metrics['energy'] = df['energy']
+    force_norm_mean = df.apply(lambda row: np.mean([np.sqrt(fx**2 + fy**2 + fz**2) for fx, fy, fz in
+                                                    zip(row['fx'], row['fy'], row['fz'])]), axis=1)
+    metrics['force_norm_average'] = force_norm_mean
+
+    x0s = df['x'][0]
+    y0s = df['y'][0]
+    z0s = df['z'][0]
+
+    displacement = df.apply(lambda row: [(x - x0) ** 2 + (y - y0) ** 2 + (z - z0) ** 2 for x, y, z, x0, y0, z0 in
```

Comment:
if I understand this correctly, this is actually the square displacement, $|{\bf x -  x}_0|^2$